extensions/kafka-streams/runtime/src/main/java/io/quarkus/kafka/streams/runtime/KafkaStreamsTopologyManager.java

```diff
@@ -189,56 +188,56 @@ private static String asString(List<InetSocketAddress> addresses) {
                 .collect(Collectors.joining(","));
     }
 
-    void onStart(@Observes StartupEvent ev) {
-        if (executor == null) {
-            return;
+    void onStop(@Observes ShutdownEvent ev) {
+        if (streams != null) {
+            LOGGER.debug("Stopping Kafka Streams pipeline");
+            streams.close();
         }
+    }
 
-        String bootstrapServersConfig = asString(runtimeConfig.bootstrapServers);
-
-        Properties streamsProperties = getStreamsProperties(properties, bootstrapServersConfig, runtimeConfig);
-
-        if (kafkaClientSupplier.isUnsatisfied()) {
-            streams = new KafkaStreams(topology.get(), streamsProperties);
-        } else {
-            streams = new KafkaStreams(topology.get(), streamsProperties, kafkaClientSupplier.get());
+    @Produces
+    @Singleton
```

Comment:
I checked on a small test, a Singleton bean seems to be not created if the bean is not injected anywhere.
Can someone confirms this behavior, I searched in CDI/Arc I did not find anything.
In our application we are not injecting (yet) a KafkaStreams instance, because we don't need it, so this is not gonna work.
Can you make sure it's working (or not) in that case : not injecting the bean somewhere.

Reply:
I found the doc : https://quarkus.io/guides/cdi-reference#lazy_by_default

` By default, CDI beans are created lazily, when needed.`
`A bean with a pseudo-scope (@Dependent and @Singleton ) is created when injected.`